src/wasi/WASI.cpp

```diff
@@ -214,6 +214,30 @@ void WASI::random_get(ExecutionState& state, Value* argv, Value* result, Instanc
     result[0] = Value(uvwasi_random_get(WASI::g_uvwasi, buf, length));
 }
 
+void WASI::args_get(ExecutionState& state, Value* argv, Value* result, Instance* instance)
+{
+    uvwasi_size_t argc;
+    uvwasi_size_t bufSize;
+    uvwasi_args_sizes_get(WASI::g_uvwasi, &argc, &bufSize);
+
+    char** uvArgv = reinterpret_cast<char**>(get_memory_pointer(instance, argv[0], argc * sizeof(char*)));
```

Comment:
I don't think this is correct. char* pointers can be 32 or 64 bit long depending on the cpu, while the output is 32 bit long offsets.

Reply:
I have created a `TemporaryData` class in the #307 patch which should help.